onnxruntime/core/framework/plugin_ep_stream.cc

```diff
@@ -3,10 +3,40 @@
 
 #include "core/framework/plugin_ep_stream.h"
 #include "core/framework/error_code_helper.h"
+#include "core/session/abi_logger.h"
+
+using namespace ::onnxruntime::logging;
+
+#define LOG_AND_RETURN_IF_ORT_ERROR(fn, logger)                                              \
+  do {                                                                                       \
+    OrtStatus* _status = (fn);                                                               \
+    if (_status != nullptr) {                                                                \
+      LOGS(logger, ERROR) << "Plug-in EP Error: [" << OrtApis::GetErrorCode(_status) << "] " \
+                          << OrtApis::GetErrorMessage(_status);                              \
+      OrtApis::ReleaseStatus(_status);                                                       \
+      return;                                                                                \
+    }                                                                                        \
+  } while (0)
 
 namespace onnxruntime {
 namespace plugin_ep {
 
+void Notification::WaitNotificationOnDevice(onnxruntime::Stream* stream, synchronize::Notification& notification) {
+  auto* this_ptr = static_cast<Notification*>(&notification);
+
+  LOG_AND_RETURN_IF_ORT_ERROR(this_ptr->impl_.WaitOnDevice(&this_ptr->impl_, static_cast<OrtSyncStream*>(stream)),
+                              *this_ptr->logger_.ToInternal());
+}
+void Notification::WaitNotificationOnHost(onnxruntime::Stream*, synchronize::Notification& notification) {
+  auto* this_ptr = static_cast<Notification*>(&notification);
+  LOG_AND_RETURN_IF_ORT_ERROR(this_ptr->impl_.WaitOnHost(&this_ptr->impl_), *this_ptr->logger_.ToInternal());
+}
+void Notification::Activate() {
+  LOG_AND_RETURN_IF_ORT_ERROR(impl_.Activate(&impl_), *logger_.ToInternal());
+}
+void Stream::Flush() {
+  LOG_AND_RETURN_IF_ORT_ERROR(impl_.Flush(&impl_), *logger_.ToInternal());
```

Comment:
What is the reason for updating the `logger_` member to be a `OrtLogger`? Maybe I missed it, but it seems like we're always just converting it back to a `logging::Logger` on use via `ToInternal()`.

Reply:
I did that to not have to forward declare the internal logger in the header. If needed i can also forward declare it and store a `Logger*` in the class.